plugins/org.jboss.tools.playground.bower/src/org/jboss/tools/playground/bower/internal/preferences/BowerPreferencePage.java

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *       Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.jboss.tools.playground.bower.internal.preferences;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jface.preference.DirectoryFieldEditor;
+import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.jboss.tools.playground.bower.internal.Activator;
+import org.jboss.tools.playground.bower.internal.BowerConstants;
+
+/**
+ * @author "Ilya Buziuk (ibuziuk)"
+ */
+public class BowerPreferencePage extends FieldEditorPreferencePage implements IWorkbenchPreferencePage {
+	public static final String PAGE_ID = "org.jboss.tools.playground.bower.internal.preferences.BowerPreferencesPage"; //$NON-NLS-1$
+	private BowerHomeFieldEditor bowerEditor;
+	
+	public BowerPreferencePage() {
+		super(GRID);
+		setPreferenceStore(Activator.getDefault().getPreferenceStore());
+		setDescription("npm Settings for Bower Support"); //$NON-NLS-1$
+	}
+	
+	@Override
+	public void init(IWorkbench workbench) {
+	}
+	
+	@Override
+	public boolean performOk() {
+		super.performOk();
+		String filePath = bowerEditor.getTextControl(getFieldEditorParent()).getText();
+		BowerPreferenceHolder.setNodeLocation(new File(filePath.trim()).getAbsolutePath());
+		return true;
+	}
+
+	@Override
+	protected void createFieldEditors() {
+		bowerEditor = new BowerHomeFieldEditor("npm Home", "npm Location", getFieldEditorParent()); //$NON-NLS-1$ //$NON-NLS-2$
+		addField(bowerEditor);
+	}
+	
+	private static class BowerHomeFieldEditor extends DirectoryFieldEditor {
+		
+		public BowerHomeFieldEditor(String key, String label, Composite composite) {
+			super(key, label, composite);
+			setEmptyStringAllowed(true);
+		}
+		
+		@Override
+		protected boolean doCheckState() {
+			String filename = getTextControl().getText();
+			filename = filename.trim();
+			if (filename.isEmpty()) {
+				this.getPage().setMessage("A location for the npm must be specified", IStatus.WARNING); //$NON-NLS-1$
```

Comment:
not sure what you are asking about but yes, I think being able to point to custom location for npm is useful in case my eclipse is not running in a enviornment that dont have npm configured.

what I'm not sure about is if npm will even run correctly based solely on running npm directly...